### PR TITLE
Add issue templates and workflows

### PR DESCRIPTION
**Pull request by mvlassis**
Ref: https://github.com/canonical/bundle-kubeflow/issues/1148

This PR creates:
- Templates under `.github/ISSUE_TEMPLATES` for issue templates
- The standard workflows we have in charm repos under `.github/workflows`

Note that it is normal for the tests to fail since the actual charm doesn't exist yet.

### CODE REVIEW

File: .github/workflows/integrate.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  workflow_call:
+    secrets:
+      charmcraft-credentials:
+        required: true
+
+jobs:
+
+  lib-check:
+    name: Check libraries
+    runs-on: ubuntu-20.04
```
Comment (kimwnasptd) [inline .github/workflows/integrate.yaml:14]: Let's update this to 24.04 as well

File: .github/workflows/integrate.yaml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          fetch-depth: 0
+      - name: Check libs
+        uses: canonical/charming-actions/check-libraries@2.3.0
+        with:
+          credentials: "${{ secrets.charmcraft-credentials }}"
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+
+  lint:
+    name: Lint Check
+    runs-on: ubuntu-20.04
```
Comment (kimwnasptd) [inline .github/workflows/integrate.yaml:28]: Let's update this to 24.04 as well

File: .github/workflows/integrate.yaml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    - name: Install dependencies
+      run: sudo apt-get install python3-pip tox
+
+    - name: Lint code
+      run: tox -e lint
+
+  unit:
+    name: Unit Test
+    runs-on: ubuntu-20.04
```
Comment (kimwnasptd) [inline .github/workflows/integrate.yaml:42]: Let's update this to 24.04 as well

File: .github/workflows/integrate.yaml [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    uses: canonical/charmed-kubeflow-workflows/.github/workflows/terraform-checks.yaml@main
+    with:
+      charm-path: .
+      # Don't run this workflow since the charm goes to `Error` state when deployed on its own
+      # See: https://github.com/canonical/resource-dispatcher/issues/66
+      apply: false
+      
+  integration:
+    name: Integration Test (build and deploy)
+    runs-on: ubuntu-20.04
```
Comment (kimwnasptd) [inline .github/workflows/integrate.yaml:65]: Let's update this to 24.04 as well

File: .github/workflows/publish.yaml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+      source_branch:
+        description: Github branch from this repo to publish.  If blank, will use the default branch
+        required: false
+        default: ''
+        type: string
+
+jobs:
+  get-charm-paths:
+    name: Generate the Charm Matrix
+    runs-on: ubuntu-20.04
```
Comment (kimwnasptd) [inline .github/workflows/publish.yaml:31]: Let's update this to 24.04 as well

File: .github/workflows/publish.yaml [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+          fetch-depth: 0
+          ref: ${{ inputs.source_branch }}
+      - name: Get paths for all charms in repo
+        id: get-charm-paths
+        run: bash .github/workflows/get-charm-paths.sh
+
+
+  publish-charm:
+    name: Publish Charm
+    runs-on: ubuntu-20.04
```
Comment (kimwnasptd) [inline .github/workflows/publish.yaml:46]: Let's update this to 24.04 as well

**Review by kimwnasptd — COMMENTED** [review]
Did a pass, could you also `grep` for any other `20.04` that we might have missed?

**Comment by mvlassis** [comment]
> Did a pass, could you also `grep` for any other `20.04` that we might have missed?

Forgot to push the commit :)

File: .github/workflows/integrate.yaml [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+      if: failure()
+
+    - name: Get operator logs
+      run: kubectl logs --tail 100 -ntesting -loperator.juju.is/name=resource-dispatcher
+      if: failure()
+    
+    - name: Setup tmate session
+      uses: mxschmitt/action-tmate@v3
+      if: failure()
+      timeout-minutes: 40
```
[the remark this answers is not in the document]
Comment (mvlassis) [reply .github/workflows/integrate.yaml:102]: This is left in the `resource-dispatcher` charm where I got the template form, I should probably add this as an issue

File: .github/workflows/integrate.yaml [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    - name: Setup operator environment
+      uses: charmed-kubernetes/actions-operator@main
+      with:
+        provider: microk8s
+        channel: 1.29-strict/stable
+        juju-channel: 3.4/stable
+        microk8s-addons: "dns storage rbac metallb:10.64.140.43-10.64.140.49"
+
+    - name: Run integration tests
+      run: tox -vve integration -- --model testing
```
[the remark this answers is not in the document]
Comment (mvlassis) [reply .github/workflows/integrate.yaml:80]: The one that we will include for sure is `kubeflow-profiles-automator`, which doesn't require the namespace to be `kubeflow`. So we should probably keep this as is it, and change it if we include other charms that have this requirement during integration.

**Review by kimwnasptd — APPROVED** [review]

**Pull request merged**